core/EE_Cron_Tasks.core.php

```diff
@@ -419,16 +419,22 @@ public static function process_expired_transactions()
                 switch ($transaction->status_ID()) {
                     // Completed TXNs
                     case EEM_Transaction::complete_status_code:
-                        /** @type EE_Transaction_Processor $transaction_processor */
-                        $transaction_processor = EE_Registry::instance()->load_class('Transaction_Processor');
-                        $transaction_processor->update_transaction_and_registrations_after_checkout_or_payment(
-                            $transaction,
-                            $transaction->last_payment()
-                        );
-                        do_action(
-                            'AHEE__EE_Cron_Tasks__process_expired_transactions__completed_transaction',
-                            $transaction
-                        );
+                        // Don't update the transaction/registrations if the Primary Registration is Not Approved.
+                        $transaction->primary_registration();
+                        if ($primary_registration instanceof EE_Registration
+                            && $primary_registration->status_ID() == EEM_Registration::status_id_not_approved
```

Comment:
wait... what?

the title of the PR is:

> Don't update the Transaction/Registrations within the cron check when the primary registrant is Not Approved

but this looks like it would ONLY do so when the primary registrant **IS** "Not Approved"

shouldn't this be:

```suggestion
                            && $primary_registration->status_ID() !== EEM_Registration::status_id_not_approved
```

either way this should use type safe comparison, ie: `===` or `!==`